src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -205,6 +207,18 @@
      */
     TransformFeatureTypeCallback transformer = new TransformFeatureTypeCallback();
 
+    /**
+     * Holds the keys for all the cache having String keys. By ensuring identity allows to
+     * synchronize on id values.
+     */
+    private CanonicalSet<String> stringCacheKeys;
```

Comment:
Maybe just call it cacheKeys? And the other one, coverageCacheKeys.

Reply:
fixed